posts/urllib3-pipenv-standalone-test.rst

```diff
@@ -0,0 +1,114 @@
+This post compares the pytests of urllib3 to its nuitka-built .whl counterpart
+The results were very similar:
+	Regular pytests: ====== 3 failed, 836 passed, 456 skipped, 113 warnings in 47.54 seconds =======
+	Nuitka wheel pytests: ====== 1 failed, 838 passed, 456 skipped, 113 warnings in 47.59 seconds =======
```

Comment:
Interesting, did you look at the 2 tests that pass now, are these mistakes maybe. It could also be that Nuitka fails to pass an exception on.

Reply:
It's very weird because the exception tracebacks match exactly, although one says 3 failed but the other says 1 failed. Is this worth looking more into?

```
====== 1 failed, 838 passed, 456 skipped, 113 warnings in 48.51 seconds =======
Exception in thread Thread-85:
Traceback (most recent call last):
  File "C:\Users\Tommy\AppData\Local\Programs\Python\Python37-32\Lib\threading.py", line 917, in _bootstrap_inner
    self.run()
  File "C:\Users\Tommy\pipenv-testing\urllib3-testing\original\urllib3\dummyserver\server.py", line 158, in run
    self.server = self._start_server()
  File "C:\Users\Tommy\pipenv-testing\urllib3-testing\original\urllib3\dummyserver\server.py", line 154, in _start_server
    self.socket_handler(sock)
  File "C:\Users\Tommy\pipenv-testing\urllib3-testing\original\urllib3\test\with_dummyserver\test_socketlevel.py", line 1152, in socket_handler
    ca_certs=DEFAULT_CA,
  File "C:\Users\Tommy\AppData\Local\Programs\Python\Python37-32\Lib\ssl.py", line 1213, in wrap_socket
    suppress_ragged_eofs=suppress_ragged_eofs
  File "C:\Users\Tommy\AppData\Local\Programs\Python\Python37-32\Lib\ssl.py", line 412, in wrap_socket
    session=session
  File "C:\Users\Tommy\AppData\Local\Programs\Python\Python37-32\Lib\ssl.py", line 850, in _create
    self.do_handshake()
  File "C:\Users\Tommy\AppData\Local\Programs\Python\Python37-32\Lib\ssl.py", line 1108, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1045)
```

>

Reply:
The comparison of such outputs is something we have code for in Nuitka, `makeDiffable` deals e.g. with filename differences, object repr differences, etc. and ought to be part of a solution where you automatically compare the output.

But generally pass and fail, which ever direction, is a problem to be investigated. Saying PASS for bad reasons is never good.

Maybe we need to start automating the comparison, because it can get otherwise too much work for you manually.

Reply:
May I suggest that you add a sentence like this:

> The extra passes are suspicious, we need to look into why these happen now. To make that easy, we are going to fully automate the process and compare outputs with verbose py.test modes.

Also, you didn't link to the issue you found already, did you?

Reply:
Mind you, after all, this is about you finding stuff. The more you find, the better. :)

Reply:
Thank you! I made the changes. Will work on automating the comparison very soon.